test/unit/data_loaders/test_data_sources.py

```diff
@@ -81,25 +81,47 @@ def test_key(self, uri, key):
         assert s3_uri.key == key
 
 
-def test_get_s3_client_built_in_dataset():
+@pytest.mark.parametrize(
+    "run_region, dataset_region",
+    [
+        ("us-west-2", "us-west-2"),
+        ("ap-east-1", "us-west-2"),
+        ("us-isof-south-1", "us-isof-south-1"),
+        ("us-isof-east-1", "us-isof-south-1"),
+    ],
+)
+@patch("boto3.session.Session")
+def test_get_s3_client_built_in_dataset(mock_session_class, run_region, dataset_region):
     """
     GIVEN a built-in dataset s3 path
     WHEN get_s3_client is called
-    THEN the boto3 s3 client is created with region name "us-west-2"
+    THEN the boto3 s3 client is created with corresponding built-in dataset region name
     """
     with patch("boto3.client") as mock_client:
+        mock_instance = mock_session_class.return_value
+        mock_instance.region_name = run_region
         dataset_uri = DATASET_CONFIGS[TREX].dataset_uri
         s3_client = get_s3_client(dataset_uri)
-        mock_client.assert_called_once_with("s3", region_name="us-west-2")
+        if dataset_region == "us-isof-south-1":
```

Comment:
This would get updated to `if dataset_region in BUILTIN_DATASET_ISO_REGIONS.values()`